### DISCUSSION TITLE
How do you display/"pick-up"  custom events from Svelte components?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [storybookjs/storybook#25677] [Help]
### Summary

I'm new to Svelte and Storybook, and trying to write stories for Svelte components.

How can I make storybook listen/pick-up on custom events emitted by the Svelte component?

I dispatch custom events from Svelte components like this:

```js
import { createEventDispatcher } from 'svelte';
const dispatch = createEventDispatcher<{'selected': string | null}>();
dispatch('selected', null)
```

My story currently looks like this:
```js
import type { Meta, StoryObj } from '@storybook/svelte';
import SimpleSelectNoForm from './root.svelte';

const meta = {
    component: SimpleSelectNoForm,
    args: {
        options: [
            { value: 'a', label: 'A' },
            { value: 'b', label: 'B' },
        ],
        name: 'simple-select-no-form',
    },
} satisfies Meta<SimpleSelectNoForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};

```

I tried using the `on` and `import { action } from '@storybook/addon-actions';`, but no output so far.

**Comment by bertdecock** [comment]
I'd like to know the same thing. @alexbjorlig did you find a way in the meantime?

**Comment by bertdecock** [comment]
For anyone having the same question, I found an answer.

In the args of your meta or story, add a property event_{name} and assign a mock function to it. Then the mock function can be verified in, for example, a play function.

In the case above, the following should work:

``` typescript
import type { Meta, StoryObj } from '@storybook/svelte';
import SimpleSelectNoForm from './root.svelte';
import { fn } from "@storybook/test";

const selectedCallback = fn();

const meta = {
    component: SimpleSelectNoForm,
    args: {
        options: [
            { value: 'a', label: 'A' },
            { value: 'b', label: 'B' },
        ],
        name: 'simple-select-no-form',
        event_selected: selectedCallback,
    },
} satisfies Meta<SimpleSelectNoForm>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {};
```

As for storybook picking the events up, I found that that works out of the box with Storybook 8. They can be viewed in the actions tab of the addon pane.
